### DISCUSSION TITLE
Lesson 4: ganache-cli is now just ganache

### DISCUSSION THREAD

**Discussion by cordair** [smartcontractkit/full-blockchain-solidity-course-py#1125] [General]
Hello. 
I ran into trouble trying to install ganache-cli with yarn.
Apparently the recommended way of doing this is (https://www.npmjs.com/package/ganache-cli)
$ npm install ganache

Or with sudo if you want to install it globally.
I'm running Debian Linux.

**Comment by cromewar** [comment]
Hello @cordair 
Well a easy way to solve this is using sudo, but you may be aware is not recommendable. Installing npm packages with sudo privileges creates a security breach on your machine.

So for me the best way to proceed is using NVM (you would have to uninstall your current npm installation)
Refer to the official github to know how to install and use NVM: 
https://github.com/nvm-sh/nvm

**Reply by n4n0b1t3** [reply]
Can you try setting up everything in a virtual environment? In this post here, I described everything you would need up to lesson 7.
https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/1109
Please let me know if this is adaptable to the Linux environment. If you had to make some changes, I would love to hear about it so I might include a Linux setup in my article.

**Reply by cromewar** [reply]
@n4n0b1t3 the command line on debian is most likely the same as on mac as they are both based on UNIX, so this should work on linux without problem, I've tested on Manjaro (arch linux) and it works (with a little workarounds)

**Reply by n4n0b1t3** [reply]
Attention: I will have to edit this post since some example directory names have dots in.their.names. This will lead to errors downstream. I am just testing this right now.

**Reply by cordair** [reply]
I deleted nodejs and npm from my system and I installed everything by using NVM.
Thanks.
